### ISSUE TITLE
Problem with creating collections on 1.30.5

### ISSUE THREAD

**Issue by dzhumshoot** [dani-garcia/vaultwarden#4555]
Hello! 
After updating from version 1.30.1 to 1.30.5 new collections are not created from the page https://myvault.com/#/vault when I click "New->Collection" and Vaultwarden is logging me out.

But if I go to the page https://myvault.com/#/organizations/uuid/vault the collections are created.

After rolling back to version 1.30.1 the problem is not reproduced.

I am the owner of the organization, the rights have not changed.

**Comment by BlackDex** [comment]
What is reported in the logs?

**Comment by dzhumshoot** [comment]
[2024-05-15 14:10:07.854][request][INFO] GET /api/organizations/9e12447b-6758-4162-b290-58fb28c6181e/collections/details
[2024-05-15 14:10:07.854][request][INFO] GET /api/organizations/9e12447b-6758-4162-b290-58fb28c6181e/users
[2024-05-15 14:10:07.855][request][INFO] GET /api/organizations/9e12447b-6758-4162-b290-58fb28c6181e/groups
[2024-05-15 14:10:07.878][auth][ERROR] Unauthorized Error: You need to be a Manager, Admin or Owner to call this endpoint
[2024-05-15 14:10:07.878][vaultwarden::api::core::organizations::_][WARN] Request guard `ManagerHeadersLoose` failed: "You need to be a Manager, Admin or Owner to call this endpoint".
[2024-05-15 14:10:07.878][auth][ERROR] Unauthorized Error: You need to be a Manager, Admin or Owner to call this endpoint
[2024-05-15 14:10:07.878][vaultwarden::api::core::organizations::_][WARN] Request guard `ManagerHeadersLoose` failed: "You need to be a Manager, Admin or Owner to call this endpoint".
[2024-05-15 14:10:07.878][response][INFO] (get_groups) GET /api/organizations/<org_id>/groups => 401 Unauthorized
[2024-05-15 14:10:07.878][response][INFO] (get_org_users) GET /api/organizations/<org_id>/users?<data..> => 401 Unauthorized
[2024-05-15 14:10:07.909][auth][ERROR] Unauthorized Error: You need to be a Manager, Admin or Owner to call this endpoint
[2024-05-15 14:10:07.909][vaultwarden::api::core::organizations::_][WARN] Request guard `ManagerHeadersLoose` failed: "You need to be a Manager, Admin or Owner to call this endpoint".
[2024-05-15 14:10:07.909][response][INFO] (get_org_collections_details) GET /api/organizations/<org_id>/collections/details => 401 Unauthorized

On the page https://myvault.com/#/vault in selected Organization i'm Owner.

**Comment by dzhumshoot** [comment]
When I click "New->Collection" from https://myvault.com/#/vault Vaultwarden is logging me out.
![image](https://github.com/dani-garcia/vaultwarden/assets/89910240/180e0f76-3e3a-433f-83ba-2b7c1cc53dbd)

**Comment by BlackDex** [comment]
Are you sure your user is a `Manager`, `Admin` or `Owner`?
This message can not happen if you are one of those.

**Comment by BlackDex** [comment]
Check the `/admin` interface and verify if the user has one of those rights.

**Comment by dzhumshoot** [comment]
Yes, I am sure. On the old version 1.30.1, collections are created under this user in this organization, there are no changes in my rights when updating.
And the problem occurs precisely when creating from the page https://myvault.com/#/vault
If I go to the page https://myvault.com/#/organizations/DVP-Security/vault the collections are created.

![image](https://github.com/dani-garcia/vaultwarden/assets/89910240/d8e551ad-4377-4ffa-903b-380c811fd921)

**Comment by BlackDex** [comment]
Also, please try the `testing` tagged image and see if that solves your issue. Be sure to create a backup of your database first btw.

**Comment by BlackDex** [comment]
Very strange, I'm not able to reproduce this with 1.30.5.
I can create a collection just fine via the main vault.

Same when using the `testing` tagged version.

**Comment by BlackDex** [comment]
I also just checked, the endpoint used to create the collection is exactly the same for both locations. So that shouldn't be an issue.
Maybe your reverse proxy is doing some magic stuff, or a browser extension is doing something? I don't know.

But running this via a local docker container with a new fresh database it works fine, and i tested it with an old (maybe even with some issues) database, and it also works fine there.

Also try a different browser and/or Private/Incognito mode.

**Comment by dzhumshoot** [comment]
Found the problem. In one of the organizations I had the role "User". The role was changed to "Manager" and collections began to be created from main vault page: https://myvault.com/#/vault
But it worked on the old version 1.30.1. Is this how it should be on 1.30.5?

This seems to be the wrong behavior, during the collection creation stage I select the organization. And it would be more correct not to display those in which I have the “User” role at the stage of creating a new collection, as it worked in the old version.

**Comment by BlackDex** [comment]
It shouldn't be different between those versions. The only thing i can think off is that you might selected the wrong organization? 

I do agree that UI wise it could be better in some cases. But that isn't something which is the main goal of this project. 

I'm going to close this one as it seems to be fixed now with the correct rights.

**Comment by dzhumshoot** [comment]
Just create several organizations, in one of which you will have the "User" role. And try to create a collection while on the page https://myvault.com/#/vault by clicking "New->Collection, you will understand what I mean. At this stage I do not select an organization, I log in and immediately try to create a collection.
The window for creating a new collection should open, as it did in the old version 1.30.1.
![image](https://github.com/dani-garcia/vaultwarden/assets/89910240/458e5bf0-59c2-4d05-ac7a-dd6e404b7ab1)

**Comment by dzhumshoot** [comment]
Please open this issue because there is a problem, the cause has been found, but there is no solution.

**Comment by stefan0xC** [comment]
Can you test with the `testing` image? As far as I've looked into it the bug seems to be caused by the web-vault version (`web-v2024.1.2`) and has already been fixed by switching to a newer version.

**Comment by dzhumshoot** [comment]
I deployed the test installation first to 1.30.5. The test user has the "User" role in one of the two test organizations (test_org2). Behavior is slightly different from a prod installation, but the essence is the same: under this user on the http://localhost/#/vault page, I click “New->Collection”, the collection creation window opens and when you select the organization “test_org2” Vaultwarden is logging me out.
Then I updated to the "testing" version. On the page http://localhost/#/vault I click “New->Collection”, a window for creating a collection opens; I don’t have the organization “test_org2” at all (and that’s correct).
![image](https://github.com/dani-garcia/vaultwarden/assets/89910240/0c4bb752-dc66-4501-9cd1-683e308e5abc)
The problem is not reproduced in the "testing" version. But on version 1.30.5 there is a problem. Is there any way to solve this?
